moha/hamiltonians.py

```diff
@@ -162,6 +162,21 @@ def generate_one_body_integral(self, basis: str, dense: bool):
         # check if elements in connectivity matrix are float
         elif np.all([isinstance(elem[2], float)
                     for elem in self.connectivity]):
+            if self.orbital_overlap is not None:
+                # Assuming self.orbital_overlap should be a 2D array
+                if hasattr(
+                        self,
+                        'orbital_overlap') and self.orbital_overlap.ndim == 2:
```

Comment:
it looks to me that check `if hasattr(
                         self,
                         'orbital_overlap')` is a redundant, since right after that we are checking if dimensionality is equal to 2. After initialization, the object type is None, so we may just substitute this expression by `if isintstance(self.orbital_overlap, np.ndarray)`.